internal/pkg/db/mongo/export.go

```diff
@@ -14,51 +14,73 @@
 package mongo
 
 import (
-	"github.com/edgexfoundry/edgex-go/internal/export"
 	"github.com/edgexfoundry/edgex-go/internal/pkg/db"
-
+	"github.com/edgexfoundry/edgex-go/internal/pkg/db/mongo/models"
+	contract "github.com/edgexfoundry/edgex-go/pkg/models"
 	"github.com/globalsign/mgo"
 	"github.com/globalsign/mgo/bson"
+	"github.com/google/uuid"
 )
 
 // ****************************** REGISTRATIONS ********************************
 
 // Return all the registrations
 // UnexpectedError - failed to retrieve registrations from the database
-func (mc MongoClient) Registrations() ([]export.Registration, error) {
-	return mc.getRegistrations(bson.M{})
+func (mc MongoClient) Registrations() ([]contract.Registration, error) {
+	var mrs []models.Registration
+	mrs, err := mc.getRegistrations(bson.M{})
+
+	return mapRegistrations(mrs, err)
 }
 
 // Add a new registration
 // UnexpectedError - failed to add to database
-func (mc MongoClient) AddRegistration(reg *export.Registration) (bson.ObjectId, error) {
+func (mc MongoClient) AddRegistration(r contract.Registration) (string, error) {
 	s := mc.getSessionCopy()
 	defer s.Close()
 
-	reg.Created = db.MakeTimestamp()
-	reg.ID = bson.NewObjectId()
+	registration := &models.Registration{}
+	id, err := registration.FromContract(r)
+
+	if err != nil {
+		return id, err
+	}
+
+	col := s.DB(mc.database.Name).C(db.ExportCollection)
+
+	registration.Created = db.MakeTimestamp()
 
 	// Add the registration
-	err := s.DB(mc.database.Name).C(db.ExportCollection).Insert(reg)
+	err = col.Insert(registration)
 	if err != nil {
-		return reg.ID, err
+		return id, err
```

Comment:
Returning an "empty" value for id instead is consistent with how error conditions are handled elsewhere.  If not, this `if` statement is redundant.